Source/WebInspectorUI/UserInterface/Views/SearchSidebarPanel.js

```diff
@@ -95,8 +95,22 @@ WI.SearchSidebarPanel = class SearchSidebarPanel extends WI.NavigationSidebarPan
 
         this._inputElement.select();
 
-        if (performSearch)
-            this.performSearch(this._inputElement.value, {omitFocus: true});
+        if (performSearch) {
+            if (this._inputElement.value.length >= WI.SearchSidebarPanel.MinimumQueryLengthForSearchOnFocus)
+                this.performSearch(this._inputElement.value, {omitFocus: true});
+            else {
+                // Allow the user to intentionally proceed with a potentially very slow query by pressing the Enter key.
+                // This is needed because the "change" event won't trigger on Enter if the value is already set; there is no change.
+                let handleKeydownOnce = (event) => {
+                    if (event.keyCode === WI.KeyboardShortcut.Key.Enter.keyCode)
+                        this.performSearch(this._inputElement.value);
+                }
+
+                this._inputElement.addEventListener("keydown", handleKeydownOnce, {once: true});
```

Comment:
I think by doing this, if the user presses something else before hitting Enter, we're not performing search. (E.g. type A, press Command, then hit Enter, and it won't search because this is a one-time listener.) Is that OK?

Reply:
also, im pretty sure a `"change"` event is fired whenever Enter is pressed so i think this might already happen via that listener?